web/modules/custom/jcc_newslinks/jcc_newslinks.module

```diff
@@ -109,20 +109,45 @@ function jcc_newslinks_cron() {
   $purgeMessage = count($purgeItems) . ' archived newslinks are queued for media purge';
   \Drupal::logger('jcc_newslinks')->notice($purgeMessage);
 
-  // Old newslinks item Purge (Batch of 100 nodes)
-//  $view = Views::getView('newslinks_expiration');
-//  $view->setDisplay('default');
-//  $view->execute();
-//  $view_result = $view->result;
-//
-//  if (!empty($view_result)) {
-//    $storage_handler = \Drupal::entityTypeManager()->getStorage("node");
-//    $nodes_to_delete = [];
-//    foreach ($view->result as $row) {
-//      $nodes_to_delete[] = $row->nid;
-//    }
-//    \Drupal::logger('jcc_newslinks')->notice('Purged newslinks with nids: ' . implode(', ', $nodes_to_delete));
-//    $entities = $storage_handler->loadMultiple($nodes_to_delete);
-//    $storage_handler->delete($entities);
-//  }
+  // Deleting newslinks if they are 3 months old
+  $threeMonthLimit = time() - (3 * 30 * 24 * 60 * 60); // 3 months ago
+
+  // Query to fetch nodes older than 3 months and in specific moderation states (draft, archived)
+  $queryOldNewsLinks = \Drupal::entityQuery('node')
+    ->condition('type', 'news')
+    ->condition('field_news_type.entity:taxonomy_term.name', 'NewsLink')
+    ->condition('created', $threeMonthLimit, '<')
+    ->condition('moderation_state', ['draft', 'archived'], 'IN') // Match multiple moderation states
+    ->range(0, 2000)  // Process 50 nodes per batch
+    ->execute();
+
+  // Log the number of nodes found
+  \Drupal::logger('jcc_newslinks')->notice('Found ' . count($queryOldNewsLinks) . ' old newslinks to delete.');
+
+  $node_storage = \Drupal::entityTypeManager()->getStorage('node');
```

Comment:
Nice! We have to convert to calls like these for Drupal 10+ so less to convert later.